File: 2020/day1/solve.rs
use std::fs::File;
use std::io::prelude::*;


fn part1(nums: &[u32]) -> u32 {
    for n1 in nums {
        for n2 in nums {
            if n1 + n2 == 2020 {
                return n1 * n2;
            }
        }
    }

    0
}

fn part2(nums: &[u32]) -> u32 {

    for n1 in nums {
        for n2 in nums {
            for n3 in nums {
                if n1 + n2 + n3 == 2020 {
                    return n1 * n2 * n3;
                }
            }
        }
    }

    0
}


fn main() -> std::io::Result<()> {

    let mut file = File::open("day1.txt")?;
    let mut inp = String::new();
    file.read_to_string(&mut inp)?;

    let nums: Vec<u32> = inp[..inp.len()-1].split("\n").map(|x| x.parse::<u32>().expect("parsing failed")).collect();

    let r1 = part1(&nums);
    println!("PART1 : {}", r1);

    let r2 = part2(&nums);
    println!("PART2 : {}", r2);
    Ok(())

}

File: 2021/day05/solve.py
import re

def part1(arr):
    for seg in inp.split("\n"):
        a,b,x,y = map(int, re.findall("\d+", seg))

        if not( a == x or b == y):
            continue

        while (a,b) != (x,y):
            arr[a][b] += 1
            if a < x: a+= 1
            if b < y : b += 1
            if a > x: a -= 1
            if b > y : b -= 1

        arr[a][b] += 1

    return len([i for i in sum(arr,[]) if i > 1])

def part2(arr):
    for seg in inp.split("\n"):
        a,b,x,y = map(int, re.findall("\d+", seg))
        arr[b][a] += 1
        while (a,b) != (x,y):
            if a < x: a+= 1
            if b < y : b += 1
            if a > x: a -= 1
            if b > y : b -= 1
            arr[b][a] += 1


    return len([i for i in sum(arr,[]) if i > 1])

inp = """0,9 -> 5,9
8,0 -> 0,8
9,4 -> 3,4
2,2 -> 2,1
7,0 -> 7,4
6,4 -> 2,0
0,9 -> 2,9
3,4 -> 1,4
0,0 -> 8,8
5,5 -> 8,2"""

inp = open("day05.txt").read().strip()

print("PART 1:", part1([[0 for _ in range(1000)] for _ in range(1000)]))
print("PART 2:", part2([[0 for _ in range(1000)] for _ in range(1000)]))

